server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader10to11.java

```diff
@@ -0,0 +1,270 @@
+            log.error("Unable to remove replication dir and files from " + replPath + ": " + ex);
+          }
+        }
+      }
+    } catch (IOException ex) {
+      log.error("Unable to remove replication dir and files: " + ex);
+    }
+  }
+
+  private boolean checkReplicationTableInZk(final InstanceId iid, final ZooReaderWriter zrw) {
+    try {
+      String path = buildRepTablePath(iid);
+      return zrw.exists(path);
+    } catch (KeeperException ex) {
+      throw new IllegalStateException("ZooKeeper error - cannot determine replication table status",
+          ex);
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("interrupted reading replication state from ZooKeeper", ex);
+    }
+  }
+
+  /**
+   * To protect against removing replication information if replication is being used and possible
+   * active, check the replication table state in Zookeeper to see if it is ONLINE (active) or
+   * OFFLINE (inactive). If the state node does not exist, then the status is considered as OFFLINE.
+   *
+   * @return true if the replication table state is OFFLINE, false otherwise
+   */
+  private boolean checkReplicationOffline(final InstanceId iid, final ZooReaderWriter zrw) {
+    try {
+      String path = buildRepTablePath(iid) + ZTABLE_STATE;
+      byte[] bytes = zrw.getData(path);
+      if (bytes != null && bytes.length > 0) {
+        String status = new String(bytes, UTF_8);
+        return TableState.OFFLINE.name().equals(status);
+      }
+      return false;
+    } catch (KeeperException ex) {
+      throw new IllegalStateException("ZooKeeper error - cannot determine replication table status",
+          ex);
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("interrupted reading replication state from ZooKeeper", ex);
+    }
+  }
+
+  /**
+   * Utility method to build the ZooKeeper replication table path. The path resolves to
+   * {@code /accumulo/INSTANCE_ID/tables/+rep}
+   */
+  static String buildRepTablePath(final InstanceId iid) {
+    return ZooUtil.getRoot(iid) + ZTABLES + "/" + REPLICATION_ID.canonical();
+  }
+
+  private void deleteReplicationTableZkEntries(ZooReaderWriter zrw, InstanceId iid) {
+    String repTablePath = buildRepTablePath(iid);
+    try {
+      zrw.recursiveDelete(repTablePath, ZooUtil.NodeMissingPolicy.SKIP);
+    } catch (KeeperException ex) {
+      throw new IllegalStateException(
+          "ZooKeeper error - failed recursive deletion on " + repTablePath, ex);
+    } catch (InterruptedException ex) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException("interrupted deleting " + repTablePath + " from ZooKeeper",
+          ex);
+    }
+  }
+
+  private void deleteReplicationConfigs(ZooReaderWriter zrw, InstanceId iid, PropStore propStore) {
+    List<PropStoreKey<?>> ids = getPropKeysFromZkIds(zrw, iid);
```

Comment:
Is this just removing them from the cache? It seems that these will be deleted from ZK later when we delete everything from ZK for this table. So, I'm not sure if this is strictly necessary to do separately. And, deleting the configs from ZK may cause the per-table configuration to enter a strange state, where its configuration now is the default per-table configuration or that set in the site configuration file or in the accumulo namespace config in ZK... which may not be what we want, because it could cause all sorts of things to change for the table that's supposed to be offline.

Reply:
No, it is more that removing them from the cache - `propStore.removeProperties`  updates ZooKeeper.  The issue is when there are replication iterator configs specified in the the configuration - particularly the metadata table. But this is also checking the other ZooKeeper configs to make sure that if the replication iterator was set it is removed.

In 2.1, the ZooKeeper config for the metadata table has these replication entries:

```
root@uno> config -t accumulo.metadata -f replc
-----------+-------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------
SCOPE      | NAME                                            | VALUE
-----------+-------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------
table      | table.iterator.majc.replcombiner .............. | 9,org.apache.accumulo.server.replication.StatusCombiner
table      | table.iterator.majc.replcombiner.opt.columns .. | stat
table      | table.iterator.minc.replcombiner .............. | 9,org.apache.accumulo.server.replication.StatusCombiner
table      | table.iterator.minc.replcombiner.opt.columns .. | stat
table      | table.iterator.scan.replcombiner .............. | 9,org.apache.accumulo.server.replication.StatusCombiner
table      | table.iterator.scan.replcombiner.opt.columns .. | stat
-----------+-------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------

```

And, the StatusCombiner, if present, will cause 3.0 code to fail because they have been removed from the code base. 

And at this point, only the metadata and root tables are online.

Not sure what to do if replication iterators have been set in the system config file - that's a manual removal - but if present they would likely cause the manager to go offline, if the upgrade sequence could complete.  The easiest way is for the upgrade instruction to include removing any replication configuration values with 2.1 down , but before starting 3.0.  There will likely be manual steps to completely remove replication remnants  - but functionally they will not prevent Accumulo from running.